girder_worker_utils/decorators.py

```diff
@@ -1,121 +1,33 @@
-from inspect import getdoc
-try:
-    from inspect import signature
-except ImportError:  # pragma: nocover
-    from funcsigs import signature
-import six
+import click
 
+from .command import Command
+from .input import Input
+from .output import Output
+from .types import String
 
-class MissingDescriptionException(Exception):
-    """Raised when a function is missing description decorators."""
-
-
-class MissingInputException(Exception):
-    """Raised when a required input is missing."""
-
-
-def get_description_attribute(func):
-    """Get the private description attribute from a function."""
-    func = getattr(func, 'run', func)
-    description = getattr(func, '_girder_description', None)
-    if description is None:
-        raise MissingDescriptionException('Function is missing description decorators')
-    return description
-
-
-def argument(name, data_type, *args, **kwargs):
-    """Describe an argument to a function as a function decorator.
-
-    Additional arguments are passed to the type class constructor.
-
-    :param str name: The parameter name from the function declaration
-    :param type: A type class derived from ``girder_worker_utils.types.Base``
-    """
-    if not isinstance(name, six.string_types):
-        raise TypeError('Expected argument name to be a string')
-
-    if callable(data_type):
-        data_type = data_type(name, *args, **kwargs)
-
-    def argument_wrapper(func):
-        func._girder_description = getattr(func, '_girder_description', {})
-        args = func._girder_description.setdefault('arguments', [])
-        sig = signature(func)
-
-        if name not in sig.parameters:
-            raise ValueError('Invalid argument name "%s"' % name)
-
-        data_type.set_parameter(sig.parameters[name], signature=sig)
-        args.insert(0, data_type)
-
-        def call_item_task(inputs, outputs={}):
-            args, kwargs = parse_inputs(func, inputs)
-            return func(*args, **kwargs)
-
-        def describe():
-            return describe_function(func)
-
-        func.call_item_task = call_item_task
-        func.describe = describe
 
+def task(name=None, cls=Command, **attrs):
```

Comment:
i'm wondering if there is a way to provide this functionality through the ```girder_worker.app.app.task``` decorator?  Otherwise functions will have to look like this:

```python
from girder_worker.app import app

@task()
@task_input('-n', type=types.Integer(min=1), required=True)
@task_output('value', type=types.Integer())
@app.task(...)
def fibonacci(n):
    pass
```

Reply:
actually i take it back,  does this task decorator belong in this PR at all? is it strictly necessary for the item_task functionality to work? See my comment below:

Reply:
I haven't addressed the integration with girder_worker yet.  We could probably combine the two `@task` decorators, but for click to work correctly we **need** a decorator on the top that serves the same purpose of the `click.command()` decorator.  That's where click actually does all of it's magic.